.gitignore

```diff
@@ -1,8 +1,8 @@
-*.o
-applet/
-*~
-*.orig
-*.rej
-*.bak
-*.DS_Store
-*.idea
+// All of the OS, IDE, compiler stuff has been deleted
+// They are needed, but you should have those in
+// your global .gitignore rather than in
+// a per-project file such as this
+
+// Our automatic versioning scheme generates the following file
+// NEVER put it in the repository
+_Version.h
```

Comment:
While it may be more appropriate to make a global .gitignore file, these patterns are included for the convenience of the forgetful – that is to say, most of us. I actually do think it's appropriate to have these patterns in this file, as a gatekeeper to this repository, even if it is redundant to what the user has in their ~/.gitignore.